src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const STATE_COPY = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(STATE_COPY, extraData);
```

Comment:
```suggestion
        STATE_COPY = {...STATE_COPY, ...extraData}```

Reply:
When I fixed this, I got errors
  28:20  error  'key' is assigned a value but never used  no-unused-vars
  
  `'use strict';

/**
 * @param {Object} state
 * @param {Object[]} actions
 *
 * @return {Object[]}
 */
function transformStateWithClones(state, actions) {
  let STATE_COPY = { ...state };
  const result = [];

  for (const action of actions) {
    const { type, extraData, keysToRemove } = action;

    switch (type) {
      case 'addProperties':
        STATE_COPY = {
          ...STATE_COPY, ...extraData,
        };
        break;
      case 'removeProperties':
        for (const key of keysToRemove) {
          delete STATE_COPY[key];
        }
        break;
      case 'clear':
        for (const key in STATE_COPY) {
          STATE_COPY = {};
        }
        break;
    }

    result.push({
      ...STATE_COPY,
    });
  }

  return result;
}

module.exports = transformStateWithClones;
`